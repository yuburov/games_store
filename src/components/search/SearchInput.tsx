import style from "./searchInput.module.scss";

export interface ISearchInputProps {
  value: string;
  onChangeText: (event: React.ChangeEvent<HTMLInputElement>) => void;
}

function SearchInput({ value, onChangeText }: ISearchInputProps): ReturnType<React.FC> {
  return (
    <div className={style.container}>
      <input className={style.searchInput} type="text" value={value} onChange={onChangeText} placeholder="Search" />
    </div>
  );
}

export default SearchInput;
