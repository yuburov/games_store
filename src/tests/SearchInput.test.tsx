import renderer from "react-test-renderer";
import SearchInput from "../components/search/SearchInput";

describe("SearchInput component tests", () => {
  test("should render input for searching", () => {
    const component = renderer.create(<SearchInput value="" onChangeText={(e) => e.target} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
