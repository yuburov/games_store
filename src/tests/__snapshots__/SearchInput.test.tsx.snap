// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput component tests should render input for searching 1`] = `
<div>
  <input
    onChange={[Function]}
    placeholder="Search"
    type="text"
    value=""
  />
</div>
`;
